samza-kafka/src/main/scala/org/apache/kafka/clients/consumer/KafkaConsumerConfig.java

```diff
@@ -0,0 +1,193 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.apache.kafka.clients.consumer;
+
+import java.util.Map;
+import java.util.Properties;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.samza.SamzaException;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.ConfigException;
+import org.apache.samza.config.JobConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Option;
+
+
+/**
+ * The configuration class for KafkaConsumer
+ */
+public class KafkaConsumerConfig extends ConsumerConfig {
+
+  public static final Logger LOG = LoggerFactory.getLogger(KafkaConsumerConfig.class);
+
+  private static final String PRODUCER_CLIENT_ID_PREFIX = "kafka-producer";
+  private static final String CONSUMER_CLIENT_ID_PREFIX = "kafka-consumer";
+  private static final String ADMIN_CLIENT_ID_PREFIX = "samza-admin";
+  private static final String SAMZA_OFFSET_LARGEST = "largest";
+  private static final String SAMZA_OFFSET_SMALLEST = "smallest";
+  private static final String KAFKA_OFFSET_LATEST = "latest";
+  private static final String KAFKA_OFFSET_EARLIEST = "earliest";
+  private static final String KAFKA_OFFSET_NONE = "none";
+
+  /*
+   * By default, KafkaConsumer will fetch ALL available messages for all the partitions.
+   * This may cause memory issues. That's why we will limit the number of messages per partition we get on EACH poll().
+   */
+  private static final String DEFAULT_KAFKA_CONSUMER_MAX_POLL_RECORDS = "100";
+
+  private KafkaConsumerConfig(Properties props) {
+    super(props);
+  }
+
+  /**
+   * Create kafka consumer configs, based on the subset of global configs.
+   * @param config
+   * @param systemName
+   * @param clientId
+   * @param injectProps
+   * @return KafkaConsumerConfig
+   */
+  public static KafkaConsumerConfig getKafkaSystemConsumerConfig(Config config, String systemName, String clientId,
+      Map<String, String> injectProps) {
+
+    final Config subConf = config.subset(String.format("systems.%s.consumer.", systemName), true);
+
+    final String groupId = getConsumerGroupId(config);
+
+    final Properties consumerProps = new Properties();
+    consumerProps.putAll(subConf);
+
+    consumerProps.setProperty(ConsumerConfig.GROUP_ID_CONFIG, groupId);
+    consumerProps.setProperty(ConsumerConfig.CLIENT_ID_CONFIG, clientId);
+
+    //Kafka client configuration
+
+    // Disable consumer auto-commit because Samza controls commits
+    consumerProps.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
```

Comment:
Since we already manage offsets in samza through checkpoint topic, we should not allow users to turn auto-commit on to manage checkpoints in kafka(when using the kafkaSystemConsumer). Kafka-autocommit is unnecessary and could potentially create confusion given that we have our own mechanism to rewind/fast-forward the checkpointed offsets(with checkpoint tool). 

(Ex: What if the user turns on kafka auto-commit, stops the job and  updates offsets stored in kafka. After this user would expect samza to resume from the updated offset in kafka. But the job would resume with the checkpointed offset stored in samza checkpoint topic)

I think we should log the warning here if auto-commit is turned on and document in configuration table that it should not be turned on.